src/ViewModels/src/TasExplorer/TasExplorerViewModel.cs

```diff
@@ -27,15 +30,29 @@ public class TasExplorerViewModel : AbstractViewModel, ITasExplorerViewModel
         private readonly IServiceProvider _services;
         private readonly IThreadingService _threadingService;
         private readonly IErrorDialog _dialogService;
+        private readonly IDataPersistenceService _dataPersistenceService;
 
         public TasExplorerViewModel(IServiceProvider services)
             : base(services)
         {
-            _dialogService = services.GetRequiredService<IErrorDialog>();
             _services = services;
+            _dialogService = services.GetRequiredService<IErrorDialog>();
             _threadingService = services.GetRequiredService<IThreadingService>();
+            _dataPersistenceService = services.GetRequiredService<IDataPersistenceService>();
 
-            TasConnection = null;
+            string existingSavedConnectionName = _dataPersistenceService.ReadStringData(ConnectionNameKey);
+            string existingSavedConnectionAddress = _dataPersistenceService.ReadStringData(ConnectionAddressKey);
+
+            if (existingSavedConnectionName != null && existingSavedConnectionAddress != null)
```

Comment:
Consider rewriting negated boolean expressions without negation.  Human brain has to work a little harder to do negation.

This expression can be rewritten as
```
if (existingSavedConnectionName == null || existingSavedConnectionAddress == null)
{
    TasConnection = null;
}
else
{
    ...
```

Reply:
👍 changed